File: LeetCode/LeetCode-CN-CPP/2020七月每日一题/718. 最长重复子数组.cpp
class Solution {
public:
    int findLength(vector<int>& A, vector<int>& B) {
        int n = A.size() , m = B.size();
        vector<vector<int>> dp(n+1, vector<int>(m+1, 0));
        int ans = 0;
        for(int i = 0; i < A.size(); i++) {
            for(int j = 0; j < B.size(); j++) {
                dp[i+1][j+1] = A[i] == B[j] ? dp[i][j] + 1 : 0;
                ans = max(dp[i+1][j+1], ans);
            }
        }
        return ans;
    }
};

File: LeetCode/LeetCode-CN-JAVA/2020七月每日一题/718. 最长重复子数组.java
class Solution {
    public int findLength(int[] A, int[] B) {
        int ans = 0;
        for(int i = 0; i < A.length; i++) {
            int maxlen = 0;
            for(int j = 0, k = i; j < B.length && k < A.length; j++, k++) {
                if(B[j] == A[k]) {
                    maxlen++;
                    ans = Math.max(ans, maxlen);
                } else {
                    maxlen = 0;
                }
            }
        }
        for(int i = 0; i < B.length; i++) {
            int maxlen = 0;
            for(int j = 0, k = i; j < A.length && k < B.length; j++, k++) {
                if(A[j] == B[k]) {
                    maxlen++;
                    ans = Math.max(ans, maxlen);
                } else {
                    maxlen = 0;
                }
            }
        }
        return ans;
    }
}